plugins/pipelines/index.js

```diff
@@ -115,7 +115,7 @@ const pipelinesPlugin = {
                     throw boom.notFound(`Pipeline ${pipelineId} does not exist`);
                 }
 
-                if (!pipeline.scmRepo || !pipeline.scmRepo.private) {
+                if (!pipeline.scmRepo || !pipeline.scmRepo.private || (pipeline.settings && pipeline.settings.public)) {
```

Comment:
can you add unit test which covers this case, looks like coverage decreased after this change